text/0202-events-and-logs-api.md

```diff
@@ -0,0 +1,172 @@
+
+Note: This could, for example, be implemented by storing any mutable configuration in the LoggerProvider and having Logger implementation objects have a reference to the LoggerProvider from which they were obtained. If configuration must be stored per-Logger (such as disabling a certain Logger), the Logger could, for example, do a look-up with its name+version+schema_url+event_domain+include_trace_context in a map in the LoggerProvider, or the LoggerProvider could maintain a registry of all returned Loggers and actively update their configuration if it changes.
+
+The effect of associating a Schema URL with a Logger MUST be that the telemetry emitted using the Logger will be associated with the Schema URL, provided that the emitted data format is capable of representing such association.
+
+### Logger
+
+The Logger is responsible for creating Events and Logs.
+
+Note that Loggers should not be responsible for configuration. This should be the responsibility of the LoggerProvider instead.
+
+#### Logger operations
+
+The Logger MUST provide functions to:
+
+- Create an `Event` and emit it to the processing pipeline.
+  - The API MUST accept an event name as a parameter. The event name provided should be inserted as an attribute with key `event.name`. Care MUST be taken to not override or delete this attribute while the `Event` is created.  This function MAY be named `logEvent`.
+- Create a `Log Record` and emit it to the processing pipeline.
+  - This function MAY be named `logRecord`.
+  - The intended users of this API is Log Appenders.
+
+### Usage
+
+```java
+OpenTelemetry openTelemetry = OpenTelemetry.noop();
+Logger logger = openTelemetry.getLogger("my-scope");
+
+// Using the convenience method to log an event directly
+logger.logEvent("network-changed", 
+                 Attributes.builder().put("type", "wifi").build());
+
+// Using the event builder to log an event
+logger.eventBuilder("page-navigated").build().setAttribute("url", "http://foo.com/bar#new").emit();
+
+// Using the logRecord builder to log a record
+Logger logger = openTelemetry.getLogger("another-scope");
+logger.logRecordBuilder().build().setBody("I am a log message").emit();
+
+```
+
+### Usage in Client-side telemetry
+
+```java
+public void addBrowserEvent(String name, Attributes attributes) {
+   Logger logger = openTelemetry.getLogger("my-scope", "1.0", "browser");
+   logger.logEvent(name, attributes);
+}
+
+public void addMobileEvent(String name, Attributes attributes) {
+   Logger logger = openTelemetry.getLogger("my-scope", "1.0", "mobile");
+   logger.logEvent(name, attributes);
+}
+```
+
+### Usage in eBPF
+
+From the eBPF [demo](https://youtu.be/F1VTRqEC8Ng?t=233), it looks like they have chosen to put the Event data in the Body field of LogRecord instead of Attributes.  They might have to make the changes to move the event data to attributes to conform to this API specification.
```

Comment:
The wording of this paragraph makes it seem like a critique of eBPF. Unless you are already very familiar with the Flowmill/otel relationship this seems like a link to a random video. Are there other components which will be changed if this OTEP is accepted?

Reply:
One other component is the [k8seventsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8seventsreceiver/k8s_event_to_logdata.go#L43) that needs to add the `event.name` attribute to the events.  I am not aware of any other cases.

Reply:
Most of my comment is not addressed. The paragraph reads like a comment in a thread or an answer to a question or something. Nothing in the paragraph says anything about when the event API might be useful for someone from an eBPF perspective and how it is different than alternatives.

Reply:
Can probably just omit this paragraph entirely.

Reply:
Yes, I removed it. Apparently, Flowmill is moving to using OTel metrics to report their telemetry data and will not be Log data model.